### PR TITLE
vp-1126_add_authorization

### PR DESCRIPTION
**Pull request by AliveMen**
add auth layer to js shopping cart

### CODE REVIEW

File: VirtoCommerce.JavaScriptShoppingCart.Model/Model/Security/SecurityConstants.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            public const string CanReadCart = "cart:read";
+            public const string CanCreateCart = "cart:create";
+            public const string CanUpdateCart = "cart:update";
+            public const string CanDeleteCart = "cart:delete";
+            public const string CanCreateOrder = "order:create";
+            public const string CanReadOrder = "order:read";
+            public const string CanReadOrderPrices = "order:read_prices";
+            public const string CanUpdateOrder = "order:update";
+
+            public static readonly IEnumerable<string> AllPermissions = new[] { CanCreateCart, CanReadCart, CanUpdateCart, CanDeleteCart, CanCreateOrder, CanReadOrder, CanReadOrderPrices, CanUpdateOrder };
```
Comment (yecli) [inline VirtoCommerce.JavaScriptShoppingCart.Model/Model/Security/SecurityConstants.cs:26]: Could you use specific permissions from the cart and order modules, its constants?
Do not want to redefine them here.

File: VirtoCommerce.JavaScriptShoppingCart.Web/Scripts/security/auth.js [hunk 2]
```diff
@@ -135,18 +153,29 @@
...
+        if (user) {
+            angular.extend(authContext, user);
+            authContext.userLogin = user.userName;
+            authContext.fullName = user.userLogin;
+            authContext.isAuthenticated = user.userName != null;
+            authContext.userId = user.id;
+            authContext.memberId = user.memberId;
+        } else {
+            authContext.userLogin = undefined;
+            authContext.isAuthenticated = false;
```
Comment (yecli) [inline VirtoCommerce.JavaScriptShoppingCart.Web/Scripts/security/auth.js:166]: Probably need to clear `memberId` too

File: VirtoCommerce.JavaScriptShoppingCart.Model/Model/Security/SecurityConstants.cs (hunk 1 above)
Comment (AliveMen) [reply VirtoCommerce.JavaScriptShoppingCart.Model/Model/Security/SecurityConstants.cs:26]: We can't reach cart constants from out module.

Comment (yecli) [reply VirtoCommerce.JavaScriptShoppingCart.Model/Model/Security/SecurityConstants.cs:26]: Indeed, it is still in Web projects in v2.

**Review by yecli — APPROVED** [review]

**Pull request merged**